File: app/src/main/java/io/github/wax911/retgraph/model/container/TrendingFeed.java
package io.github.wax911.retgraph.model.container;

import android.support.annotation.StringDef;

import java.util.List;

import io.github.wax911.retgraph.model.parent.Entry;

public class TrendingFeed {

    // https://api.githunt.com/graphiql feed types, represented as StringDef instead of enums
    public final static String HOT = "HOT", NEW = "NEW", TOP = "TOP";
    @StringDef({HOT,NEW, TOP})
    @interface FeedType {}

    private List<Entry> feed;

    public List<Entry> getFeed() {
        return feed;
    }
}

File: app/src/main/java/io/github/wax911/retgraph/model/container/TrendingFeed.kt
package io.github.wax911.retgraph.model.container

import androidx.annotation.StringDef
import io.github.wax911.retgraph.model.parent.Entry

data class TrendingFeed(val feed: List<Entry>?) {

    @StringDef(HOT, NEW, TOP)
    internal annotation class FeedType

    companion object {
        // https://api.githunt.com/graphiql feed types, represented as StringDef instead of enums
        const val HOT = "HOT"
        const val NEW = "NEW"
        const val TOP = "TOP"
    }
}
